script/tool/test/version_check_test.dart

```diff
@@ -196,6 +196,73 @@ void main() {
       expect(gitDirCommands[2].join(' '),
           equals('show HEAD:packages/plugin_platform_interface/pubspec.yaml'));
     });
+
+     test('Throws if first line in change log is empty', () async{
+      createFakePlugin('plugin', includeChangeLog: true, includeVersion: true);
+
+      final Directory pluginDirectory =
+          mockPackagesDir.childDirectory('plugin');
+
+      createFakePubspec(pluginDirectory, isFlutter: true, includeVersion: true, version: '1.0.1');
+      String changelog = '''
+
+## 1.0.1
+
+* Some changes.
+''';
+      createFakeCHANGELOG(pluginDirectory, changelog);
+      final Future<List<String>> output = runCapturingPrint(
+          runner, <String>['version-check', '--base_sha=master']);
+      await expectLater(
+        output,
+        throwsA(const TypeMatcher<FormatException>()),
+      );
+    });
+
+    test('Throws if versions in changelog and pubspec do not match', () async{
+      createFakePlugin('plugin', includeChangeLog: true, includeVersion: true);
+
+      final Directory pluginDirectory =
+          mockPackagesDir.childDirectory('plugin');
+
+      createFakePubspec(pluginDirectory, isFlutter: true, includeVersion: true, version: '1.0.1');
+      String changelog = '''
+## 1.0.2
+
+* Some changes.
+''';
+      createFakeCHANGELOG(pluginDirectory, changelog);
+      final Future<List<String>> output = runCapturingPrint(
+          runner, <String>['version-check', '--base_sha=master']);
+      await expectLater(
+        output,
+        throwsA(const TypeMatcher<Error>()),
+      );
+    });
+
+    test('Success if CHANGELOG and pubspec versions match', () async{
+      createFakePlugin('plugin', includeChangeLog: true, includeVersion: true);
+
+      final Directory pluginDirectory =
+          mockPackagesDir.childDirectory('plugin');
+
+      createFakePubspec(pluginDirectory, isFlutter: true, includeVersion: true, version: '1.0.1');
+      String changelog = '''
+## 1.0.1
+
+* Some changes.
+''';
+      createFakeCHANGELOG(pluginDirectory, changelog);
+      final List<String> output = await runCapturingPrint(
+          runner, <String>['version-check', '--base_sha=master']);
+      await expect(
+        output,
+        containsAllInOrder([
+          'Checking that plugin has matching version in CHANGELOG and pubspec',
+          'No version check errors found!'
+        ]),
+      );
+    });
   });
```

Comment:
Please include a test where the version matches an older entry in the CHANGELOG (an actual bug that's happened).

Reply:
done